### ISSUE TITLE
Validation throws a 500 rather than 400

### ISSUE THREAD

**Issue by mikebrules** [AndrewKeig/express-validation#31]
Hi, I've stripped out all other middleware and used the example in the README, express-validation has installed Joi as a dependency so version should be okay, but it is throwing a 500 on invalidate req object rather than the expected 400. Here is the test case;

`     

```
 //app.js  
router.use(bodyParser.json());
router.post('/:version/:publisher/:id/:assetType', 
validate(storyValidator),
(req, res) => {

  const content = req.content;
  const assetsToQueuePromises = [];
  const assetsToRedisPromises = [];
  const storyIds = {topCardId:undefined, cardIds:[]};
  res.send(200);
   .....
 })

 //storyValidator
 const Joi = require('joi'); 
   module.exports = {
   body: {
     email: Joi.string().email().required(),
     password: Joi.string().regex(/[a-zA-Z0-9]{3,30}/).required()
   }
 };

 //test.js
  it.only('should validate a card correctly when all data is present',(done) => {
    request(app)
    .post('/v1/ft/765890/card')
    .send({
      email: "andrew.keig@gmail.com",
      password: ""})
    .expect(400)  //actually throws 500
    .end( (err, res) => {
      console.log(err);
      done();
    })

   `
```

Thanks in advance - what am I doing wrong. Using 1.0.0

**Comment by mikebrules** [comment]
Haha, Just seen this on the README
     `
       
     // error handler, required as of 0.3.0

     app.use(function(err, req, res, next){
       res.status(400).json(err);
     });
    `
    I'm guessing that is it!

**Comment by colthreepv** [comment]
Sorry, this is a recurring question, I will re-open and look for better documentation options.
If someone happens to have a better solution, can propose it here, or via PR

**Comment by mantagen** [comment]
Getting seemingly strange behaviour, whereby if a joi validation schema item has `.required()`, the 400 is sent immediately.
If no `.required()`, but request still fails on that item, an error is thrown but after the handler has been entered.
Is this intended behaviour?

**Comment by AndrewKeig** [comment]
Closing, released 3.0
